frontend/catalyst/jax_tracer.py

```diff
@@ -689,45 +694,137 @@ def trace_quantum_function(
         # (1) - Classical tracing
         quantum_tape = QuantumTape()
         with EvaluationContext.frame_tracing_context(ctx) as trace:
+            # WARNING: Assumption happening here:
+            # out_tree_promise holds the output pytree structure for the
+            # original quantum function being trace. We assume that
+            # all transforms will preserve this type as the output.
+            #
+            # But this might not be strictly necessary.
             wffa, in_avals, out_tree_promise = deduce_avals(f, args, kwargs)
             in_classical_tracers = _input_type_to_tracers(trace.new_arg, in_avals)
             with QueuingManager.stop_recording(), quantum_tape:
                 # Quantum tape transformations happen at the end of tracing
                 ans = wffa.call_wrapped(*in_classical_tracers)
-            out_classical_tracers_or_measurements = [
-                (trace.full_raise(t) if isinstance(t, DynamicJaxprTracer) else t) for t in ans
-            ]
 
+            out_classical_tracers = []
+            out_measurements = []
+            out_classical_tracers_or_measurements = []
+            for ret_val in ans:
+                if isinstance(ret_val, DynamicJaxprTracer):
+                    ret_val_variable = trace.full_raise(ret_val)
+                    out_classical_tracers.append(ret_val_variable)
+                else:
+                    ret_val_variable = ret_val
+                    out_measurements.append(ret_val_variable)
+                out_classical_tracers_or_measurements.append(ret_val_variable)
+
```

Comment:
Maybe just a comment that it mimicks the qnode logic.

Reply:
Added comment here: https://github.com/PennyLaneAI/catalyst/pull/280/commits/7e5cda7a8208c64c727fb79a0b2f089b364a9cc8